src/model/utils/insertcontent.js

```diff
@@ -13,24 +13,61 @@ import Element from '../element';
 import Range from '../range';
 import log from '@ckeditor/ckeditor5-utils/src/log';
 import DocumentSelection from '../documentselection';
+import Selection from '../selection';
 
 /**
  * Inserts content into the editor (specified selection) as one would expect the paste
  * functionality to work.
  *
+ *		// Insert text at current document selection position.
+ *		model.change( () => {
+ *			insertContent( model, new Text( 'x' ) ); // equal to insertContent( model, new Text( 'x' ), editor.model.document.selection );
+ *		} );
+ *
+ *		// Insert text at some position - model's selection will not be modified.
+ *		model.change( () => {
+ *			insertContent( model, new Text( 'x' ), new Position( doc.getRoot(), [ 2 ] ) );
+ *		} );
+ *
+ * If an instance of {module:engine/model/selection~Selection} is passed as `selectable` it will be modified
+ * to the insertion selection (equal to a range to be selected after insertion).
+ *
+ *		// Insert text replacing given selection instance.
+ *		const selection = new Selection( new Position( doc.getRoot(), [ 2 ] ), new Position( doc.getRoot(), [ 5 ] ) );
+ *
+ *		model.change( () => {
+ *			insertContent( model, new Text( 'x' ), selection );
+ *		} );
+ *
+ *		// The selection contains a range to be selected - ie can be used to set selection.
+ *		model.change( ( writer ) => {
+ *			writer.setSelection( selection );
+ *		} );
```

Comment:
I think these docs would be more useful for `model.insertContent` because this the API users will use.